soroban-env-host/src/host.rs

```diff
@@ -2516,4 +2516,19 @@ impl VmCallerCheckedEnv for Host {
         let res = HostVec::from_vec(self.0.budget.clone(), outer)?;
         Ok(self.add_host_object(res)?.into())
     }
+
+    fn fail_with_status(
+        &self,
+        vmcaller: &mut VmCaller<Self::VmUserState>,
+        status: Status,
+    ) -> Result<RawVal, Self::Error> {
+        if status.is_type(ScStatusType::ContractError) {
+            Err(self.err_status_msg(status, "failing with contract-provided status code"))
```

Comment:
Lets use the term `contract error status code`, because the status type is contract error.
```suggestion
            Err(self.err_status_msg(status, "failing with contract error status code"))
```

Reply:
integrated with rebase